src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -126,7 +126,11 @@ public IRowToRowMapper GetRowToRowMapper(Schema inputSchema)
             return (IRowToRowMapper)Scorer.ApplyToData(Host, new EmptyDataView(Host, inputSchema));
         }
 
-        protected void SaveModel(ModelSaveContext ctx)
+        void ICanSaveModel.Save(ModelSaveContext ctx) => SaveModel(ctx);
+
+        private protected abstract void SaveModel(ModelSaveContext ctx);
+
+        protected void SaveModelCore(ModelSaveContext ctx)
```

Comment:
Also, I think `SaveModelCore` should at least be `private protected`, probably best-friended as well. (Actually I might think that prediction transformers should be best-friended on `SaveModel` as well, but I don't insist on it.

Reply:
I made the `SaveModel` BestFriend, I think that's what you meant. 

---
In reply to: [254475268](https://github.com/dotnet/machinelearning/pull/2431#discussion_r254475268) [](ancestors = 254475268)